### PR TITLE
Fix Navigation

### PR DESCRIPTION
**Pull request by Graf009**
The component [HeaderRow] children forced to wrapped to the [Navigation], it is bad because I can not correctly embed [Button], my solution replace {this._renderNav ()} to {this.props.children}

### CODE REVIEW

**Comment by Graf009** [comment]
Now I make an example http://www.getmdl.io/templates/dashboard/index.html on react-mdl

**Comment by Graf009** [comment]
I'm still working on it

**Comment by Graf009** [comment]
I finished

File: src/layout/HeaderRow.js [hunk 1]
```diff
@@ -1,22 +1,11 @@
...
-        var hasChildren = React.Children.count(this.props.children);
-        if(!hasChildren) return;
-
-        return (
-            <Navigation>
-                {this.props.children}
-            </Navigation>
-        );
+        title: PropTypes.string,
+        children: PropTypes.object
```
Comment (tleunen) [inline src/layout/HeaderRow.js]: `children` not needed, you can safely remove it :)

File: src/layout/Layout.js [hunk 2]
```diff
@@ -32,5 +32,6 @@ export default mdlUpgrade(Layout);
 export Header from './Header';
 export Drawer from './Drawer';
 export HeaderRow from './HeaderRow';
+export Navigations from './Navigation';
```
Comment (tleunen) [inline src/layout/Layout.js]: Should be `Navigation` (without 's')

**Comment by tleunen** [comment]
Thanks @Graf009. I added a few comments.
I think it's a good idea to put the Navigation as a separate component so the end user can do what he wants inside the header/drawer. 

Could you please read my comments and then merge all your commit into one? (`git rebase -i` should help you)

**Comment by Graf009** [comment]
Ready

File: src/layout/HeaderRow.js [hunk 3]
```diff
@@ -28,7 +16,7 @@ class HeaderRow extends React.Component {
             <div className={classes} {...otherProps}>
                 {title ? <span className="mdl-layout-title">{title}</span> : null}
                 <div className="mdl-layout-spacer"></div>
```
Comment (tleunen) [inline src/layout/HeaderRow.js]: Should we replace this with the new `<Spacer />` or remove it?
I don't see any use-cases where we would not put a spacer between the title and the children/navigation...

File: src/layout/Navigation.js [hunk 4]
```diff
@@ -1,11 +1,19 @@
-import React from 'react';
+import React, { PropTypes } from 'react';
 import classNames from 'classnames';
 import cloneChildren from '../utils/cloneChildren';
 
 class Navigation extends React.Component {
+	static propTypes = {
+        className: PropTypes.string
```
Comment (tleunen) [inline src/layout/Navigation.js]: indentation is not right here

**Comment by tleunen** [comment]
Thanks so much! Please rebase, fix the indentation and then I'll merge :)

**Comment by Graf009** [comment]
I can not do rebase can help me?

**Comment by tleunen** [comment]
Done, thanks!
There are a few other things I wanna add to the project before releasing a new version. Hope that's ok

**Pull request closed without merging**